src/IsScrolled/IsScrolled.js

```diff
@@ -0,0 +1,92 @@
+import { Component, createRef } from "react";
+import PropTypes from "prop-types";
+import ResizeObserver from "resize-observer-polyfill";
+
+import onScrollFn from "../utils/onScroll";
+import onResizeFn from "../utils/onResize";
+
+export default class IsScrolled extends Component {
+  static propTypes = {
+    children: PropTypes.func,
+    render: PropTypes.func,
+  }
+
+  static defaultProps = {
+    children: null,
+    render: null,
+  }
+
+  state = {
+    isScrolledTo: {
+      left: true,
+      top: true,
+      right: false,
+      bottom: false,
+    },
+    isScrollable: {
+      x: false,
+      Y: false,
+    },
+  }
+
+  containerRef = createRef();
+
+  contentRef = createRef();
+
+  observer = null;
+
+  componentDidMount() {
+    const {
+      containerRef: {
+        current: $container,
+      },
+      contentRef: {
+        current: $content,
+      },
+    } = this;
+
+    const {
+      onScroll,
+      onResize,
+    } = this;
+
+    if (!$container) {
```

Comment:
Why can `$container` be not truthy? Maybe this should be console.errored (in dev mode).

Reply:
Oh definitely it should at least console.warned because It most likely shows that the developer does not know how to use your package (did not assign refs)

Reply:
Well actually this if is used there as the ref isn't assigned immediately. Only after a rerender it gets assigned. So not giving an if would always throw an error that `$container is undefined`

Reply:
Ok as it appears so I was wrong and it works perfectly without it 🤔No idea why that happened.